test/Numerics/DGMethods/compressible_navier_stokes_equations/shared_source/grids.jl

```diff
@@ -0,0 +1,187 @@
+# Description 
+Computes a DiscontinuousSpectralElementGrid as specified by a product domain
+# Arguments
+-`Ω`: A product domain object
+# Keyword Arguments 
+-`elements`: A tuple of integers ordered by (Nx, Ny, Nz) for number of elements
+-`polynomialorder`: A tupe of integers ordered by (npx, npy, npz) for polynomial order
+-`FT`: floattype, assumed Float64 unless otherwise specified
+-`topology`: default = StackedBrickTopology
+-`mpicomm`: default = MPI.COMM_WORLD
+-`array`: default = ClimateMachine.array_type()
+-`brickbuilder`: default = uniform_brick_builder, 
+  brickrange=uniform_brick_builder(Ω, elements)
+# Return 
+A DiscontinuousSpectralElementGrid object
+"""
+function DiscontinuousSpectralElementGrid(
+    Ω::ProductDomain;
+    elements = nothing,
+    polynomialorder = nothing,
+    FT = Float64,
+    mpicomm = MPI.COMM_WORLD,
+    array = ClimateMachine.array_type(),
+    topology = StackedBrickTopology,
+    brick_builder = uniform_brick_builder,
+)
+
+    if elements == nothing
+        error_message = "Please specify the number of elements as a tuple whose size is commensurate with the domain,"
+        error_message *= " e.g., a 3 dimensional domain would need a specification like elements = (10,10,10)."
+        error_message *= " or elements = (vertical = 8, horizontal = 5)"
+
+        @error(error_message)
+        return nothing
+    end
+
+    if polynomialorder == nothing
+        error_message = "Please specify the polynomial order as a tuple whose size is commensurate with the domain,"
+        error_message *= "e.g., a 3 dimensional domain would need a specification like polynomialorder = (3,3,3)."
+        error_message *= " or polynomialorder = (vertical = 8, horizontal = 5)"
+
+        @error(error_message)
+        return nothing
+    end
+
+    dimension = ndims(Ω)
+
+    if (dimension < 2) || (dimension > 3)
+        error_message = "SpectralElementGrid only works with dimensions 2 or 3. "
+        error_message *= "The current dimension is " * string(ndims(Ω))
+
+        println("The domain is ", Ω)
+        @error(error_message)
+        return nothing
+    end
+
+    elements = convention(elements, Val(dimension))
+    if ndims(Ω) != length(elements)
+        @error("Incorrectly specified elements for the dimension of the domain")
+        return nothing
+    end
+
+    polynomialorder = convention(polynomialorder, Val(dimension))
+    if ndims(Ω) != length(polynomialorder)
+        @error("Incorrectly specified polynomialorders for the dimension of the domain")
+        return nothing
+    end
+
+    brickrange = brick_builder(Ω, elements, FT = FT)
+
+    if dimension == 2
+        boundary = ((1, 2), (3, 4))
+    else
+        boundary = ((1, 2), (3, 4), (5, 6))
+    end
+
+    periodicity = periodicityof(Ω)
+
+    topl = topology(
+        mpicomm,
+        brickrange;
+        periodicity = periodicity,
+        boundary = boundary,
+        connectivity = :face,
```

Comment:
does anyone know why this causes things to segfault? @jkozdon @skandalaCLIMA @simonbyrne 

this change is needed or else the 2D tests fail (seems to make sense based on the other changes from #2071 ) but the 3D test also segfaulted without this (e.g. with `connectivity = :full`).

All of this was working perfectly fine before rebasing on top of master

Reply:
If we have both 2D and 3D examples, I would recommend using
connectivity = dimension == 3 ? :full :  :face

Reply:
can this be done automatically in the constructor? seems like a lot of work for the user to remember if you can't actually have the different combinations

Reply:
Yes, we eventually plan to support full connectivity for all cases. We will make sure to set the defaults once we add the support.